types/office-runtime/index.d.ts

```diff
@@ -13,101 +13,191 @@ Copyright (c) Microsoft Corporation
+     * Returns a Promise.
+     * @param keyValues Key-value pairs to be set. Must be strings.
+     */
+    setItems(keyValues: { [key: string]: string }): Promise<void>;
     /**
-     * Method that enables a pop up web dialog box.
-     * @param url Must be a string.
-     * @param options Optional parameter. Must be of type DisplayWebDialogOptions.
+     * Removes multiple items from storage.
+     * Returns a Promise.
+     * @param keys Keys of items to be removed. Must be an array of strings.
      */
-    function displayWebDialog(url: string, options?: DisplayWebDialogOptions): Promise<Dialog>;
+    removeItems(keys: string[]): Promise<void>;
     /**
-     * Asynchronous, global, and persistent key-value storage.
+     * Retrieves an array of all keys from storage.
+     *  Returns a Promise.
      */
-    const storage: Storage;
+    getKeys(): Promise<string[]>;
+  }
 
+  /** Object representing the dialog box. */
+  interface Dialog {
+    /** Method to close a dialog box. Returns a Promise. */
+    close(): Promise<void>;
+  }
+
+  /** Provides display options and actions a dialog box may take. */
+  interface DisplayWebDialogOptions {
+    /**
+     * Optional parameter that determines whether the dialog box displays as a popup (false) or within an IFrame (true).
+     * This setting is only applicable to custom functions running on Excel Online.
+     */
+    displayInIFrame?: boolean;
+    /**
+     * Optional parameter that defines the height of the dialog box as a percentage of the current display.
+     * For example, accepts strings such as: '50%', '50'.
+     */
+    height?: string;
     /**
-     * Asynchronous, global, and persistent key-value storage.
+     * Optional parameter that defines the width of dialog as a percentage of window.
+     * For example, accepts strings such as: '50%', '50'.
+     */
+    width?: string;
+    /**
+     * Optional callback that runs when the dialog box sends a message to its parent.
+     */
+    onMessage?: (message: string, dialog?: Dialog) => void;
+    /**
+     * Optional callback that runs when the dialog box is closed.
+     */
+    onClose?: () => void;
+    /**
+     * Optional callback that runs when the dialog box sends an error.
+     */
+    onRuntimeError?: (error: Error, dialog?: Dialog) => void;
+  }
+  /**
+   * Contains authorization related APIs.
+   * @beta
+   */
+  const auth: Auth;
+  /**
+   * Provides options for the user experience when Office obtains an access token to the add-in from AAD v. 2.0 with the getAccessTokenAsync method.
+   */
+  interface AuthOptions {
+    /**
+     * Allows Office to get an access token silectly or through interactive consent, if one is required
+     * If set to false, Office will silently try to get an access token. If it fails to do so, Office will return a descriptive error
+     * If set to true, Office will show an interactive consent UI after it fails to silently get an access token
```

Comment:
We need to add that the prompt will only allow consent to the AAD profile scope, not to any Microsoft Graph scopes.